server/src/main/java/org/opensearch/cluster/coordination/NoMasterBlockService.java

```diff
@@ -76,14 +76,24 @@
         "write",
         NoMasterBlockService::parseNoMasterBlock,
         Property.Dynamic,
+        Property.NodeScope,
+        Property.Deprecated
+    );
+    // The setting below is going to replace the above.
+    // To keep backwards compatibility, the old usage is remained, and it's also used as the fallback for the new usage.
+    public static final Setting<ClusterBlock> NO_CLUSTER_MANAGER_BLOCK_SETTING = new Setting<>(
+        "cluster.no_cluster_manager_block",
+        NO_MASTER_BLOCK_SETTING,
+        NoMasterBlockService::parseNoMasterBlock,
+        Property.Dynamic,
         Property.NodeScope
     );
 
     private volatile ClusterBlock noMasterBlock;
 
     public NoMasterBlockService(Settings settings, ClusterSettings clusterSettings) {
```

Comment:
When are we targeting updating the language in places like these?

Reply:
For the methods that published maven as Java APIs, I think there are 2 paths.
1. Replace the name for all of them in a future major version, which will can be tracked in issue https://github.com/opensearch-project/OpenSearch/issues/1684.
2. Create alternative methods and deprecate the existing methods. This is a normal path to deprecate Java APIs. Although I can make the change in separate PRs, there are too many methods with the term "master", it will be a huge effort.
I think a feasible way is to replace the "master" term in them all in a future major version, without adding @deprecate annotation and creating alternative methods. Besides, it's reasonable to choose a few common methods to follow the normal deprecation way to add alternative usages for them.

Reply:
As @mch2 noted below, I was referring to internal class and method names that include the _master_ language. I agree that incorporating those changes would make the diff rather large. Can you create an issue to track this renaming, if there isn't one already? This can be implemented/committed independent of #1684

Reply:
Looks like the `NoMasterBlockService` is published to Maven in jar file (https://opensearch.org/javadocs/1.2.4/OpenSearch/server/build/docs/javadoc/org/opensearch/cluster/coordination/NoMasterBlockService.html), so I thought it's not a internal method. However, seems there are few (or no) software is using APIs in this class.
There are so many classes like this one... I'm not sure if we should follow a normal deprecation path, or just rename.
In addition, there is a issue to track the renaming of internal class and method https://github.com/opensearch-project/OpenSearch/issues/1548, but I didn't scan and list all usages in detail.